c++/src/Compression.cc

```diff
@@ -260,31 +270,53 @@ namespace orc {
       throw std::runtime_error("Failed to reset inflate.");
     }
 
-    strm_.avail_in = static_cast<unsigned int>(bufferSize);
-    strm_.next_in = rawInputBuffer.data();
+    // reset output buffer
+    if (outputPosition != 0) {
+      BufferedOutputStream::BackUp(outputSize - outputPosition);
```

Comment:
Sorry but I don't get the intention we reset output buffer here.

Reply:
never mind, I've delete it